content/chainguard/migration/migrating-python.md

```diff
@@ -0,0 +1,124 @@
+
+```Dockerfile
+# syntax=docker/dockerfile:1
+
+FROM cgr.dev/chainguard/python:latest-dev as dev
+
+WORKDIR /flask-app
+
+RUN python -m venv venv
+ENV PATH="/flask-app/venv/bin":$PATH
+COPY requirements.txt requirements.txt
+RUN pip install -r requirements.txt
+
+FROM cgr.dev/chainguard/python:latest
+
+WORKDIR /flask-app
+
+COPY app.py app.py
+COPY --from=dev /flask-app/venv /flask-app/venv
+ENV PATH="/flask-app/venv/bin:$PATH"
+
+EXPOSE 8000
+
+ENTRYPOINT ["python", "-m", "gunicorn", "-b", "0.0.0.0:8000", "app:app"]
+```
+
+When running an application containerized with the above Dockerfile, the application should be visible on `0.0.0.0:8000`.
+
+As you can see, the primary difference in this Flask application compared to the pre-migration application is the use of a multistage build. In the initial stage, we copy our requirements into the development version of the Python Chainguard Image, initialize a virtual environment, and install needed packages with pip. In the second stage, we copy the virtual environment from the development image, copy the application from the host, set exposed port ;metadata, and run the application with the [Gunicorn](https://gunicorn.org/) WSGI server.
+
+By default, the entrypoint for the Python Chainguard Image is `python` rather than `bash`. However, if you replace the included Python binary with the virtual environment binary on the path as we do above, you should set the entrypoint explicitly, otherwise you will not have access to the packages included in your virtual environment.
+
+We recommend that you pin dependencies to specific versions in your own application. The example Flask application script linked above also enables debug mode, which should be turned off in a production scenario.
+
+You may wish to include the following environmental variables in your Dockerfile. They prevent the buffering of output and prevent the creation of cached bytecode, respectively, and can be included in both stages of the multi-stage build.
+
+```Dockerfile
+ENV PYTHONUNBUFFERED 1
+ENV PYTHONDONTWRITEBYTECODE 1
+```
+
+## Serving an Application with nginx and Docker Compose
+
+We provide an [nginx Chainguard Image](/chainguard/chainguard-images/reference/nginx/), also with low to no CVEs, that can be used as a secure and performant reverse proxy to serve your application. You can view an [example orchestration of a Flask application and nginx using Chainguard Images](https://github.com/chainguard-dev/cg-images-python-migration/tree/compose-flask-nginx) at the linked repository. The `compose.yml` file is provided as a reference below.
```

Comment:
It's not an adjective, so I don't know why it would have hyphens. 🤷 I could be persuaded, but I'll leave as-is for now.

This is how I would do it:

> It's a low-to-no-CVE image, so it's pretty cool.

> The image is pretty cool because it has low to no CVEs.

If it's a style guide thing then I'll change it. :)